### PR TITLE
Notification Detail: show custom messages

### PR DESCRIPTION
**Pull request by keithjgrant**
##### SUMMARY
Adds custom messages to the Notification Detail, if they have been customized. I also created an ArrayDetail component to clean up the display of multi-value details (such as Slack destination channels, Twilio destination numbers, etc)

Addresses #7877 

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

**Comment by unlikelyzero** [comment]
@tiagodread I think you can start adding tests for this one even in it's current state

**Comment by tiagodread** [comment]
@keithjgrant tests are written and once the PR receives dev review, I'll run the tests again and it can be merged.

File: awx/ui_next/src/screens/NotificationTemplate/NotificationTemplateDetail/NotificationTemplateDetail.jsx [hunk 1]
```diff
@@ -358,4 +371,164 @@ function NotificationTemplateDetail({ i18n, template }) {
...
+          mode="jinja2"
+          rows="6"
+          fullWidth
+        />
+      )}
+      {showMessages && (
+        <CodeDetail
+          label={i18n._(t`Workflow approved message`)}
+          value={
+            messages.workflow_approval.approved.message ||
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/NotificationTemplate/NotificationTemplateDetail/NotificationTemplateDetail.jsx]:
```suggestion
            messages?.workflow_approval?.approved?.message ||
```

@keithjgrant 
`messages.workflow_approval` can be `null` so I think we need to check before referencing its properties or use optional chaining in the various places it is referenced:

Screenshot from 2020-09-14 13-17-15

Screenshot from 2020-09-14 13-18-06

File: awx/ui_next/src/screens/NotificationTemplate/shared/hasCustomMessages.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  if (!messages) {
+    return false;
+  }
+
+  return (
+    isCustomized(messages.started, defaults.started) ||
+    isCustomized(messages.success, defaults.success) ||
+    isCustomized(messages.error, defaults.error) ||
+    isCustomized(
+      messages.workflow_approval.approved,
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/NotificationTemplate/shared/hasCustomMessages.js]:
```suggestion
      messages?.workflow_approval?.approved,
```

**Review by jakemcdermott — CHANGES_REQUESTED** [review]

File: awx/ui_next/src/screens/NotificationTemplate/NotificationTemplateDetail/NotificationTemplateDetail.jsx (hunk 1 above)
Comment (jakemcdermott) [reply awx/ui_next/src/screens/NotificationTemplate/NotificationTemplateDetail/NotificationTemplateDetail.jsx:438]: probably needed here as well: https://github.com/ansible/awx/blob/devel/awx/ui_next/src/screens/NotificationTemplate/shared/NotificationTemplateForm.jsx#L144-L170

File: awx/ui_next/src/screens/NotificationTemplate/shared/hasCustomMessages.js (hunk 2 above)
Comment (jakemcdermott) [reply awx/ui_next/src/screens/NotificationTemplate/shared/hasCustomMessages.js]: probably needed here as well: https://github.com/ansible/awx/blob/devel/awx/ui_next/src/screens/NotificationTemplate/shared/NotificationTemplateForm.jsx#L144-L170

**Review by jakemcdermott — APPROVED** [review]

**Comment by jakemcdermott** [comment]
After looking a bit further into the `null` `workflow_approval` field w/ @keithjgrant I'm starting to think this might be an unaddressed data migration issue with pre-existing notification templates and the (more recently added) workflow approval feature. It's probably simpler to just handle the `null` in the ui at this point so that's what we're doing.

**Comment by tiagodread** [comment]
@keithjgrant After clicking on the select all notifications on notification templates list isn’t possible to unselect items:

chrome-capture (5)

**Review by tiagodread — APPROVED** [review]
Tests are written 👍🏽

**Pull request merged**